apps/subscriptions/api/views.py

```diff
@@ -125,7 +126,35 @@ def activation(self, request, pk=None):
         except Exception as e:
             logger.error(f"Error {'activating' if subscription.is_activated else 'deactivating'} subscription: {extract_error_message(e)}")
             return self._create_rendered_response(
-                {"error": extract_error_message(e)},
+                {"error": "Something went wrong"},
+                status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+    
+    @CacheService.cached(tag_prefix="services:list", timeout=86400)
```

Comment:
How long do you desire this to be?, add it to the CacheService and use it anyway to maintain consistency
```suggestion
    @CacheService.cached(tag_prefix="services:list", timeout=CacheService.YOUR_TIME_OUT_HERE)
```